### PR TITLE
add db level constraint for at most one active subscription per domain

### PR DESCRIPTION
**Pull request by nickpell**
Should run without issue on all production environments, since we monitor and remove duplicate active subscriptions.

### CODE REVIEW

File: corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    dependencies = [
+        ('accounting', '0005_automatic_downgrade_adjustment_method'),
+    ]
+
+    operations = [
+        HqRunSQL(
+            """
+            CREATE UNIQUE INDEX accounting_subscription_active_subscriber
+            ON accounting_subscription(subscriber_id) WHERE (is_active = TRUE and is_hidden_to_ops = FALSE);
```
Comment (snopoke) [inline corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py]: Unless you're very sure this won't fail on any env you should do it manually beforehand and the use something like this to allow the migration to go ahead even with the index created: https://github.com/dimagi/commcare-hq/blob/c8a1d42bf1de3adace8cbebfb1559d302a05caae/corehq/util/django_migrations.py#L33-L33

File: corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            """,
+            reverse_sql=
+            """
+            DROP INDEX accounting_subscription_active_subscriber;
```
Comment (snopoke) [inline corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py]: Use `DROP INDEX IF EXISTS` to be safe

Comment (nickpell) [inline corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py]: updated

File: corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py]: updated so it can be run manually beforehand

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
Added `don't merge` label till you've had a chance to run this manually.

**Comment by nickpell** [comment]
ran on all envs

**Comment by nickpell** [comment]
@dimagi/team-commcare-hq If running this migration gives you problems locally, you'll want to delete any duplicate active subscriptions for the same domain (offline with me on how to do this)

**Pull request merged**